### PR TITLE
Fixes #2439 Misleading info in global POP PK Analysis if no parameter…

### LINKED ISSUE #2439: 11.1 Misleading info in global POP PK Analysis if no parameters can be calculated
[test11.1_PK_IV_Multi.pksim5.zip](https://github.com/Open-Systems-Pharmacology/PK-Sim/files/10074821/test11.1_PK_IV_Multi.pksim5.zip)
In the example attached, some simulations have **IV multiple** administration protocol. Apparently no global PK Parameters are calculated in such a case.
In an **individual** simulation, the global PK parameters area is just not shown in such a case:
grafik

In a **population** simulation, the message "Please re-run..." is shown all the time, which is very confusing.
Either the message should be adjusted in this particular case or the global PK parameters area should be hidden (same as in individual simulations)
grafik

### PR DESCRIPTION
**Pull request by rwmcintosh**
…s can be calculated

### CODE REVIEW

File: src/PKSim.Presentation/Presenters/Simulations/GlobalPKAnalysisPresenter.cs [hunk 1]
```diff
@@ -164,6 +165,20 @@ public bool HasParameters()
          return false;
       }
 
+      public bool CanCalculateGlobalPK()
```
Comment (rwmcintosh) [inline src/PKSim.Presentation/Presenters/Simulations/GlobalPKAnalysisPresenter.cs:168]: I ran resharper to auto-format so there's some noise in this PR. The gist is we need to calculate when a simulation contains only multiple IV applications of all compounds. If there's only single IV, oral, or a mix we want to show the global panel.

File: src/PKSim.Core/Services/PKAnalysesTask.cs [hunk 2]
```diff
@@ -648,20 +637,20 @@ private bool isMultipleOral(Simulation simulation, Compound compound)
          return allSchemaItems.Any() && allSchemaItems.All(isOral);
       }
 
-      private static bool isOral(ISchemaItem schemaItem)
+      private static bool isOral(SchemaItem schemaItem)
```
Comment (msevestre) [inline src/PKSim.Core/Services/PKAnalysesTask.cs:640]: Why this chnage from Interface to not interface?

File: src/PKSim.Presentation/Presenters/Simulations/GlobalPKAnalysisPresenter.cs (hunk 1 above)
Comment (msevestre) [reply src/PKSim.Presentation/Presenters/Simulations/GlobalPKAnalysisPresenter.cs:168]: it this how it was doen for individual simulation?

File: src/PKSim.Presentation/Presenters/Simulations/PopulationPKAnalysisPresenter.cs [hunk 3]
```diff
@@ -30,18 +30,16 @@ public class PopulationPKAnalysisPresenter : PKAnalysisPresenter<IPopulationPKAn
...
          IPKParameterRepository pkParameterRepository, IPresentationSettingsTask presentationSettingsTask,
          IGlobalPKAnalysisPresenter globalPKAnalysisPresenter)
-         : base(view, pkParameterRepository, presentationSettingsTask)
+         : base(view, pkParameterRepository, presentationSettingsTask, globalPKAnalysisPresenter)
       {
          _pkAnalysesTask = pkAnalysesTask;
          _exportTask = exportTask;
          _populationPKAnalysisToDTOMapper = populationPKAnalysisToDTOMapper;
-         _globalPKAnalysisPresenter = globalPKAnalysisPresenter;
          AddSubPresenters(_globalPKAnalysisPresenter);
```
Comment (msevestre) [inline src/PKSim.Presentation/Presenters/Simulations/PopulationPKAnalysisPresenter.cs:43]: hum...this should go in the parent presenter and be gone here I think

File: src/PKSim.Presentation/Presenters/Simulations/PopulationPKAnalysisPresenter.cs [hunk 4]
```diff
@@ -67,7 +65,9 @@ public void CalculatePKAnalyses(IPopulationDataCollector populationDataCollector
          var simulation = populationDataCollector.DowncastTo<PopulationSimulation>();
 
          //Calculate first global PK that should always be updated
-         _globalPKAnalysisPresenter.CalculatePKAnalysis(new[] {simulation});
+         _globalPKAnalysisPresenter.CalculatePKAnalysis(new[] { simulation });
```
Comment (msevestre) [inline src/PKSim.Presentation/Presenters/Simulations/PopulationPKAnalysisPresenter.cs:68]: not even sure this guy should be protected (_globalPKAnalysisPresenter)

File: tests/PKSim.Tests/Presentation/GlobalPKAnalysisPresenterSpecs.cs [hunk 5]
```diff
@@ -62,6 +66,154 @@ public void has_parameters_should_return_false()
...
+      protected override void Because()
+      {
+         _result = sut.CanCalculateGlobalPK();
+      }
+
+      protected abstract List<SchemaItem> GetFirstCompoundSchemaItems();
+   }
+
+
+   public class when_calculating_if_a_global_pk_analysis_is_possible_for_single_iv : context_for_can_calculate_global_pk_analysis
```
Comment (msevestre) [inline tests/PKSim.Tests/Presentation/GlobalPKAnalysisPresenterSpecs.cs:137]: when = > When

**Review by msevestre — CHANGES_REQUESTED** [review]

File: src/PKSim.Presentation/Presenters/Simulations/PopulationPKAnalysisPresenter.cs (hunk 3 above)
Comment (rwmcintosh) [reply src/PKSim.Presentation/Presenters/Simulations/PopulationPKAnalysisPresenter.cs:43]: Oh, I see what you mean, the other lines referencing that object can go up too

File: src/PKSim.Presentation/Presenters/Simulations/GlobalPKAnalysisPresenter.cs [hunk 6]
```diff
@@ -164,6 +165,20 @@ public bool HasParameters()
...
+      {
+         return firstSimulation.Compounds.Any(compound =>
+         {
+            var schemaItems = _protocolToSchemaItemsMapper.MapFrom(firstSimulation.CompoundPropertiesFor(compound).ProtocolProperties.Protocol);
+            return !isMultipleIV(schemaItems);
+         });
+      }
+
+      private bool isMultipleIV(IReadOnlyList<SchemaItem> schemaItems)
```
[the remark this answers is not in the document]
Comment (rwmcintosh) [reply src/PKSim.Presentation/Presenters/Simulations/GlobalPKAnalysisPresenter.cs]: It's similar. In Individual =>
`_view.GlobalPKVisible = _globalPKAnalysisPresenter.HasParameters();`

[the remark this answers is not in the document]
Comment (rwmcintosh) [reply src/PKSim.Presentation/Presenters/Simulations/GlobalPKAnalysisPresenter.cs]: Fixed

File: src/PKSim.Presentation/Presenters/Simulations/PopulationPKAnalysisPresenter.cs (hunk 3 above)
Comment (rwmcintosh) [reply src/PKSim.Presentation/Presenters/Simulations/PopulationPKAnalysisPresenter.cs]: Fixed

File: tests/PKSim.Tests/Presentation/GlobalPKAnalysisPresenterSpecs.cs (hunk 5 above)
Comment (rwmcintosh) [reply tests/PKSim.Tests/Presentation/GlobalPKAnalysisPresenterSpecs.cs]: Fixed

**Review by msevestre — APPROVED** [review]

**Pull request merged**